main.php

```diff
@@ -102,6 +102,32 @@ function fillHeader($header)
                     break;
                 }
 
+
+                if ($parameters["columns_from"] === 'header') {
+                    // ensure all slices have same headers
+                    $headers = [];
+                    $previousHeader = null;
+                    foreach ($subFinder as $slicedFilePart) {
+                        $csv = new Keboola\Csv\CsvFile(
+                            $slicedFilePart->getPathname(),
+                            $manifest["delimiter"],
+                            $manifest["enclosure"]
+                        );
+                        $header = $csv->getHeader();
+                        unset($csv);
```

Comment:
Má ten unset tady nějaký speciální důvod?

Reply:
Ano, zavírá file handle.

Reply:
Je pravda, že by se měl zavřít ve chvíli, kdy se udělá GC a v další iteraci je ta proměnná už přepsaná, ale přišlo mi fajn to mít explicitně.

Reply:
jo to jo, ale proč nepočkat až ho to zlikviduje samo? V té smyčce to zavolá destruktor hned jak jde na další CsvFile.

Reply:
Jo, asi máš pravdu.

Reply:
plati i pro PHP: https://blogs.msdn.microsoft.com/oldnewthing/20100809-00/?p=13203/

Reply:
Já z toho čtu, že by se na destruct vůbec nemělo spoléhat. A měla by být explicitní metoda na zavření.

Reply:
ano, tak nejak

Reply:
https://github.com/keboola/php-csv/issues/32